src/components/BoxForTransaction.tsx

```diff
@@ -138,49 +143,93 @@ export const BoxForTransaction = () => {
         }
 
         try {
-          dispatch({
-            type: "setState",
-            payload: {
-              loadingMessage: "Processing Transaction...",
-            },
-          })
+          await web3.eth
+            .estimateGas(transactionParameters)
+            .then(function (gasEstimate) {
+              transactionParameters.gas = gasEstimate
+            })
 
-          const response = await web3.eth.sendTransaction(transactionParameters)
+          try {
+            dispatch({
+              type: "setState",
+              payload: {
+                loadingMessage: "Processing Transaction...",
+              },
+            })
 
-          dispatch({
-            type: "setState",
-            payload: {
-              loadingMessage: null,
-            },
-          })
+            const response = await web3.eth.sendTransaction(
+              transactionParameters,
+            )
 
-          setIsTransactionDetailsModalOpen(false)
-          setIsTransactionResultModalOpen(true)
-          setTransactionResult({
-            blockHash: cleanHex(`${response.blockHash}`, 66),
-            blockNumber: `${response.blockNumber}`.replace(/[^0-9]/g, ""),
-            transactionHash: cleanHex(`${response.transactionHash}`, 66),
-          })
-          dispatch({
-            type: "setState",
-            payload: {
-              loadedFileContents: null,
-              validatedDeposits: [],
-              previouslyDepositedPubkeys: [
-                ...previouslyDepositedPubkeys,
-                ...allPubkeys,
-              ],
-            },
-          })
+            dispatch({
+              type: "setState",
+              payload: {
+                loadingMessage: null,
```

Comment:
This could actually just move into the payload on line 180 below, and then this extra dispatch can be deleted